api/types/container/secret.go

```diff
@@ -0,0 +1,12 @@
+package container
+
+import "os"
+
+type ContainerSecret struct {
```

Comment:
I did it this way to keep responsibilities separate.  The node has the secrets and define it as part of the container.  The engine checks the Container definition and creates the secrets in the tmpfs as needed.  We could change it but I think this keeps it cleanly separated.  Perhaps it shouldn't be.